clients/command-kafka/src/main/java/org/eclipse/hono/client/command/kafka/KafkaBasedInternalCommandConsumer.java

```diff
@@ -393,46 +392,47 @@ void handleCommandMessage(final KafkaConsumerRecord<String, Buffer> record) {
-                            command.getTenant(),
-                            HttpURLConnection.HTTP_NOT_FOUND));
-                } else {
-                    commandContext.release(new ServerErrorException(
-                            command.getTenant(),
-                            HttpURLConnection.HTTP_UNAVAILABLE,
-                            "error retrieving tenant configuration",
-                            t));
-                }
-            })
-            .onSuccess(tenantConfig -> {
-                commandContext.put(CommandContext.KEY_TENANT_CONFIG, tenantConfig);
-                if (commandHandler != null) {
-                    // partition index and offset here are related to the *tenant-based* topic the command was originally received in
-                    // therefore they are stored with the tenant as key
-                    final Map<Integer, Long> lastHandledPartitionOffsets = lastHandledPartitionOffsetsPerTenant
-                            .computeIfAbsent(command.getTenant(), k -> new HashMap<>());
-                    final Long lastHandledOffset = lastHandledPartitionOffsets.get(commandPartition);
-                    if (lastHandledOffset != null && commandOffset <= lastHandledOffset) {
-                        LOG.debug("ignoring command - record partition offset {} <= last handled offset {} [{}]", commandOffset,
-                                lastHandledOffset, command);
-                        TracingHelper.logError(currentSpan, "command record already handled before");
-                        commandContext.release(new CommandAlreadyProcessedException());
-                    } else {
-                        lastHandledPartitionOffsets.put(commandPartition, commandOffset);
-                        LOG.trace("using [{}] for received command [{}]", commandHandler, command);
-                        // command.isValid() check not done here - it is to be done in the command handler
-                        commandHandler.handleCommand(commandContext);
-                    }
-                } else {
-                    LOG.info("no command handler found for command [{}]", command);
-                    commandContext.release(new NoConsumerException("no command handler found for command"));
-                }
-            });
+        // partition index and offset here are related to the *tenant-based* topic the command was originally received in
+        // therefore they are stored with the tenant as key
+        final Map<Integer, Long> lastHandledPartitionOffsets = lastHandledPartitionOffsetsPerTenant
+                .computeIfAbsent(command.getTenant(), k -> new HashMap<>());
+        final Long lastHandledOffset = lastHandledPartitionOffsets.get(commandPartition);
+        if (lastHandledOffset != null && commandOffset <= lastHandledOffset) {
+            LOG.debug("ignoring command - record partition offset {} <= last handled offset {} [{}]", commandOffset,
```

Comment:
this should be guarded by a check if debug level is enabled on the logger to prevent the command object's toString() method to be invoked unnecessarily

Reply:
I've added that.